server/providers/Audible.js

```diff
@@ -6,83 +6,81 @@ class Audible {
     }
 
-    getBestImageLink(images) {
-        if (!images) return null
-        var keys = Object.keys(images)
-        if (!keys.length) return null
-        return images[keys[keys.length - 1]]
-    }
-
-    getPrimarySeries(series, publication_name) {
-        return (series && series.length > 0) ? series.find((s) => s.title == publication_name) || series[0] : null
-    }
-
     isProbablyAsin(title) {
         return /^[0-9A-Z]{10}$/.test(title)
     }
 
     asinSearch(asin) {
-        var queryObj = {
-            response_groups: 'rating,series,contributors,product_desc,media,product_extended_attrs',
-            image_sizes: '500,1024,2000'
-        };
-        var queryString = (new URLSearchParams(queryObj)).toString();
         asin = encodeURIComponent(asin);
-        var url = `https://api.audible.com/1.0/catalog/products/${asin}?${queryString}`
+        var url = `https://api.audnex.us/books/${asin}`
         Logger.debug(`[Audible] ASIN url: ${url}`)
         return axios.get(url).then((res) => {
-            if (!res || !res.data || !res.data.product || !res.data.product.authors) return []
-            return [res.data.product]
+            if (!res || !res.data || !res.data.asin) return null
+            return res.data
         }).catch(error => {
-            Logger.error('[Audible] search error', error)
+            Logger.error('[Audible] ASIN search error', error)
             return []
         })
     }
 
-    async search(title, author) {
-        if (this.isProbablyAsin(title)) {
-            var items = await this.asinSearch(title)
-            if (items.length > 0) return items.map(item => this.cleanResult(item))
+    async search(title, author, asin) {        
+        var items
+        if(asin) {
+            items = [await this.asinSearch(asin)]
+        }
+        
+        if (!items && this.isProbablyAsin(title)) {
+            items = [await this.asinSearch(title)]
         }
 
-        var queryObj = {
-            response_groups: 'rating,series,contributors,product_desc,media,product_extended_attrs',
-            image_sizes: '500,1024,2000',
-            num_results: '25',
-            products_sort_by: 'Relevance',
-            title: title
-        };
-        if (author) queryObj.author = author
-        var queryString = (new URLSearchParams(queryObj)).toString();
-        var url = `https://api.audible.com/1.0/catalog/products?${queryString}`
-        Logger.debug(`[Audible] Search url: ${url}`)
-        var items = await axios.get(url).then((res) => {
-            if (!res || !res.data || !res.data.products) return []
-            return res.data.products
-        }).catch(error => {
-            Logger.error('[Audible] search error', error)
-            return []
-        })
-        return items.map(item => this.cleanResult(item))
+        if(!items) {
+            var queryObj = {
+                response_groups: 'rating,series,contributors,product_desc,media,product_extended_attrs',
```

Comment:
If now we only need to get the ASIN from the audible api I think we can safely remove the response groups and image sizes.